extensions/amp-base-carousel/1.0/amp-base-carousel.css

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+amp-base-carousel {
+  display: block;
+  overflow: hidden;
+}
+
+amp-base-carousel > * {
+  box-sizing: border-box !important;
+  margin: 0 !important;
+  flex-shrink: 0 !important;
+  height: 100%;
```

Comment:
Hmm. This is a tricky one. IIUC, we layout the children here on AMP layer. But it's a critical part of the React component too? I.e. does the React component work correctly without these styles? 

I think there are two models here: 
1. Stretch: force children to take 100% space. Requires (more or less) stylesheet with `!important`.
2. Contain: have an internal slide wrapper that takes 100% of space, but uses centering flexbox for each slide's content.

Do we have any idea which one would work better from the component user point of view?

Reply:
Yes, I had thought about putting this in the Preact layer but left it in AMP for the following reasons:
- These are styles on the user-supplied DOM. In Preact we would have to merge styles if we want to insert them.
- Just like we leave AMP runtime to handle layout, we leave Preact publishers to specify the dimensions that they want their `BaseCarousel` and underlying slides -- IMO it's not good devX if we make them do this and only sometimes respect it. 

The original reason I ended up including these styles are to keep amp-base-carousel visually consistent between 0.1 and 1.0 for [this sample](https://github.com/ampproject/amphtml/blob/c14b507eda631475233d739e4653eb0c6009d101/test/manual/amp-base-carousel/captions.amp.html) involving captioned responsive images. I added the Preact counterpart sample in the Storybook if you want to take a look at these.

Reply:
Discussed offline: Placing styles in Preact rendered `<style>` tag. They must also stay in the AMP css file because the styles rendered in Preact do not propagate across the `<slot>` element in AMP mode.